storage/src/backend/connection.rs

```diff
@@ -363,7 +348,62 @@ impl Connection {
                 });
             }
         }
-        // TODO: check mirrors' health
+        // Check mirrors' health
+        for mirror in connection.mirrors.iter() {
+            let conn = connection.clone();
```

Comment:
It's better to extract this for loop into a new function as function `new` is already too long

Reply:
Ok, I add two new function: `start_proxy_health_thread` and `start_mirror_health_thread`.